presto-orc/src/main/java/com/facebook/presto/orc/stream/OrcInputStream.java

```diff
@@ -94,13 +94,15 @@ public OrcInputStream(
             int sliceInputPosition = toIntExact(sliceInput.position());
             int sliceInputRemaining = toIntExact(sliceInput.remaining());
             this.buffer = new byte[sliceInputRemaining];
+            bufferMemoryUsage.setBytes(sliceInputRemaining);
```

Comment:
> sliceInputRetainedSizeInBytes was already counted on line 91. For the no compression case, counting the buffer size here is double counting.

SliceInput uses some memory, as it has its own buffers. But then we are allocating a new buffer, the `this.buffer` that uses additional memory that should be accounted. 

The design here is really weird and . Basically the logic for accounting `FixedLengthSliceInput` memory resides in `OrcInputStream`, that is really weird. 

> setBytes() should be counting all retained memory for this object, but here you're only counting buffer. It is just one part of the memory used by OrcInputStream object. Other parts include instance size, temporaryBuffer size, compressedSliceInput but these are not counted.

I agree with that. It is very weird that only the `buffer` memory is accounted. Ideally we should account for the `OrcInputStream` instance as well. Do you think it can be further refactored and optimized in a separate PR?

Reply:
I think line 91's original intention was to count for the `buffer`. We can remove line 91 as the OrcInputStream doesn't hold a reference to the the input slice, and count the buffer and other objects that OrcInputStream hold reference to

Reply:
> We can remove line 91 as the OrcInputStream doesn't hold a reference to the the input slice, and count the buffer and other objects that OrcInputStream hold reference to

Unfortunately it cannot be removed. It accounts for the memory usage of the `FixedLengthSliceInput`. This is a little twisted, and probably will have to be refactored. But for this PR I would like to keep it as is

> and count the buffer and other objects that OrcInputStream hold reference to

Ok, let me refactor that

Reply:
> Unfortunately it cannot be removed. It accounts for the memory usage of the FixedLengthSliceInput. This is a little twisted, and probably will have to be refactored. But for this PR I would like to keep it as is

Actually nevermind. I can see 

> this.compressedSliceInput = EMPTY_SLICE.getInput();

So the original `sliceInput` is not retained